packages/contracts/contracts/discovery/Project.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.12;
+
+import {IController} from "./IController.sol";
+import {IProject} from "./IProject.sol";
+
+contract Project is IProject {
+    // deployed by each individual project owner, when registering
+    // must be deployed via the Controller
+    // will have a similar role as the CTND Vesting contract
+
+    address public controller;
+    address public token;
+
+    string public description;
+    uint256 public saleSupply;
+    uint256 public immutable rate;
+
+    constructor(
+        string memory _description,
+        address _token,
+        uint256 _saleSupply,
+        uint256 _rate
+    ) {
+        controller = msg.sender;
+
+        description = _description;
+        token = _token;
+        saleSupply = _saleSupply;
+        rate = _rate;
+    }
+
+    function hasTokens() private pure returns (bool) {
+        return true;
+    }
+
+    function isApproved() private pure returns (bool) {
+        return true;
+    }
+
+    function isReady() external pure returns (bool) {
+        return hasTokens() && isApproved();
+    }
+
+    modifier isProjectInBatch() {
```

Comment:
I know I wrote it like this, but I think we need to make it clearer. the names sounds like we're asking "is it in any batach?", but the implementation is actually checking "is this in the batch that is myself?"

since this is a modifier, something like `onlyBatch` would make sense